### ISSUE TITLE
Between Minetest Versions, Mapgen Changes Cause Huge Rock Walls

### ISSUE THREAD

**Issue by poikilos** [poikilos/EnlivenMinetest#281]
Reproduce the issue:
* Create world in a MT 5 2017 git version
  - seed: 52534
  - map meta (unchanged from default): [map_meta.zip](https://github.com/poikilos/EnlivenMinetest/files/3231161/map_meta.zip)
* load the world and walk around
* Upgrade MT 5 to a later version or to MT 6
* Load the world again and walk to the parts not yet generated

Characteristics of World with Seed 52534:
* In test world B sent to upstream Final Minetest today, the forest at spawn differs from the world as spawned with the same seed using Minetest 5.0.0-git from 2017. The world was used into 2018.
* The world, as originally generated around 2017, had a plains biome at spawn, and that crescent-shaped canyon lake was created by the worldgen of that version.

Background (regarding MT 5's changes):
If there is any way to solve that issue, then worlds converted from MT 5 to Final Minetest wouldn't have the characteristic huge rock walls caused by changing anything that affects mapgen enough. This is mostly a multiplayer issue, for worlds that don't do map resets. I'm not sure whether this is possible to fix, as rock walls even appeared at random commits as I discovered after doing git pull and rebuilt from .net's GitHub code at various times during that period. I never have seen the .net site offering any warning about this ~~, but that may expected since I was using a git version~~. :edit: The change is unexpected since the map_meta (including mapgen version) is the same (server owners don't expect huge mapgen artifacts making the world hard to navigate and making minetestmapper maps display the blocky artifacts) unless they fixed it by the next release (preventing the artifacts when using same map_meta between stable releases), which seems doubtful since it seems totally out of control.

Even if you could just somehow prevent those rock walls between versions of Final Minetest, that would assist with the issue at least for worlds created using Final Minetest. The solution may involve:
* honoring map_meta better, such as:
  - putting changes to mapgen in a new mapgen version, such as beyond 7
  - putting changes to mapgen in parameters of map_meta that are not added to it except in new worlds

**Comment by slopsbucket** [comment]
A very important issue. In the mainstream version Paramat won't leave the mapgen alone, he keeps changing it and changing it and changing it again.

That was one of the major issues when we first started working on Final Minetest - we settled on a mapgen that did everything we wanted and stuck with it. So with Final Minetest and Bucket Game all map seeds have been  perfectly compatible with each other for more than 4 years now. This is part of the stability that needs to be offered to allow serious consideration of Minetest as a long lasting platform for world builders.

Cheers,
Andrew.
